alicloud/config.go

```diff
@@ -399,6 +402,57 @@ func (c *Config) getAuthCredential(stsSupported bool) auth.Credential {
 	return credentials.NewAccessKeyCredential(c.AccessKey, c.SecretKey)
 }
 
+func (client *AliyunClient) AccountId() (string, error) {
+	client.accountIdMutex.Lock()
+	defer client.accountIdMutex.Unlock()
```

Comment:
I am sorry, I can not understand why you added a new field accountIdMutex? I think just use a logic is ok: if accountId is empty, account will be get from sts.

Reply:
I use the mutex in order to be thread-safe: as you can see the account_id is loaded only when necessary and then cached. The reason I want to be safe is because Terraform can execute several actions in parallel.